components/ui/location-combobox.tsx

```diff
@@ -0,0 +1,187 @@
+
+// TODO: Replace with a DB call to get the cached regions
+const cachedLocations: APIRegionArrayFormatted = [
+  {
+    region_id: "602703",
+    type: "CITY",
+    regionNames: {
+      fullName: "San Jose, California, United States",
+      shortName: "San Jose",
+      displayName: "San Jose, California, United States",
+      primaryDisplayName: "San Jose",
+      secondaryDisplayName: "California, United States",
+      lastSearchName: "San Jose, California, United States",
+    },
+    coordinates: {
+      lat: "37.3354",
+      long: "-121.891907",
+    },
+    country: {
+      name: "United States",
+      domain: "US",
+    },
+  },
+  {
+    region_id: "4409939",
+    type: "AIRPORT",
+    regionNames: {
+      fullName:
+        "San Jose (SJC - Norman Y. Mineta San Jose Intl.), California, United States",
+      shortName: "SJC",
+      displayName: "San Jose Airport",
+      primaryDisplayName: "Norman Y. Mineta San Jose Intl.",
+      secondaryDisplayName: "San Jose, California, United States",
+      lastSearchName: "San Jose (SJC - Norman Y. Mineta San Jose Intl.)",
+    },
+    coordinates: {
+      lat: "37.369739",
+      long: "-121.929225",
+    },
+    country: {
+      name: "United States",
+      domain: "US",
+    },
+  },
+  {
+    region_id: "3177",
+    type: "CITY",
+    regionNames: {
+      fullName: "San José, San José Province, Costa Rica",
+      shortName: "San José",
+      displayName: "San José, Costa Rica",
+      primaryDisplayName: "San José",
+      secondaryDisplayName: "San José Province, Costa Rica",
+      lastSearchName: "San José, Costa Rica",
+    },
+    coordinates: {
+      lat: "9.93286",
+      long: "-84.079559",
+    },
+    country: {
+      name: "Costa Rica",
+      domain: "CR",
+    },
+  },
+];
+
+export default function LocationCombobox() {
+  const [value, setValue] = useContext(RegionContext);
+  const [open, setOpen] = useState(false);
+  const [ops, setOps] = useState(cachedLocations);
+  const [query, setQuery] = useState("");
+
+  const findRegion = () => {
+    handleFindRegion(query, ops, setOps);
+  };
+  const dropdownRef = React.useRef<HTMLDivElement>(null);
+  const handleBlur = (e: React.FocusEvent) => {
+    // Check if focus is still within the dropdown
+    if (!dropdownRef.current?.contains(e.relatedTarget as Node)) {
+      setOpen(false);
+    }
+  };
+
+  return (
+    <Command
+      ref={dropdownRef}
+      onFocus={() => setOpen(true)}
+      onBlur={handleBlur}
+      className={`relative ${
+        open ? "rounded-lg rounded-b-none" : "rounded-lg"
+      } text-neutral dark:text-gray-100 dark:bg-gray-800 border-2 border-primary focus:outline-none focus:ring-2 focus:ring-blue-500`}
+    >
+      <CommandInput
+        placeholder="Search locations..."
+        onValueChange={setQuery}
+        onKeyDown={(e) => {
+          if (e.key === "Enter") findRegion();
+          else if (e.key === "Escape") setOpen(false);
+        }}
+      />
```

Comment:
I think this is the input/search box.

Clicking on a selected location should autofill the searchbox with the selected region's name. I believe query in `Line 144` should have the fully displayname of the region.